File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    q = []
    for i in scoville:
        heapq.heappush(q, i)
    
    while q[0] < K:
        heapq.heappush(q, heapq.heappop(q) + (heapq.heappop(q) * 2))
        answer+=1
    	
        if len(q) == 1 and q[0] < K:
            return -1
    return answer

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        Queue<Integer> pq = new PriorityQueue<>();
        
        for(int s : scoville){
            pq.add(s);
        }
        
        while (pq.peek() < K){
            if(pq.size() == 1){ //모든 음식의 스코빌 지수가 K 미만인 경우
                return -1;
            }
            pq.add(pq.poll() + pq.poll() * 2);
            answer+=1;
        }
        return answer;
    }
}